jupyterkernelgen/jupyterkernelgen.py

```diff
@@ -0,0 +1,278 @@
+    print(f"{colors.HEADER}ENTER KERNEL NAME{colors.ENDC}: ", end="", flush=True)
+    kernel_name = sys.stdin.readline().strip()
+    print()
+
+    return kernel_name
+
+def create_kernel_dir(kernel_name: str) -> str:
+    """
+    Create the necessary directory for installing the kernel
+
+    :param kernel_name: the name of the kernel to generate
+    :return:            the path to the new directory
+    """
+    print(f"{colors.HEADER}CREATING KERNEL DIRECTORY...{colors.ENDC}")
+    path = None
+    try:
+        # replace ~ with the path to the user's home directory
+        path = os.path.expanduser(f"~/.local/share/jupyter/kernels/{kernel_name}")
+        if os.path.exists(path):
+            print(f"{colors.FAIL}Environment with that name already exists. Exiting...{colors.ENDC}", file=sys.stderr)
+            raise JupyterKernelGenException
+        os.makedirs(path, exist_ok=True)
+    except OSError as e:
+        print(f"{colors.FAIL}Failed to create directory: {e}{colors.ENDC}", file=sys.stderr)
+        raise JupyterKernelGenException
+    print()
+    return path
+
+def create_kernel_helper_script(path: str, conda_env: str) -> None:
+    """
+    Create a script that launches the kernel
+
+    :param path:        the path to install the kernel to
+    :param conda_env:   the path to the conda environment to launch with
+    """
+    print(f"{colors.HEADER}CREATING KERNEL HELPER SCRIPT...{colors.ENDC}")
+    source = f"""#!/bin/bash
+
+source activate {conda_env}
+exec "$@"
+"""
+    try:
+        f = open(path + "/kernel-helper.sh", "w")
+        f.write(source)
+        f.close()
+    except OSError as e:
+        print(f"{colors.FAIL}failed to write to {path}/kernel-helper.sh: {e}{colors.ENDC}", file=sys.stderr)
+        raise JupyterKernelGenException
+    print()
+
+def create_kernel_json(path, kernel_name):
+    """
+    Create the json defining the kernel
+
+    :param path:        the path to install the kernel to
+    :param kernel_name: the name of the kernel being installed
+    """
+    print(f"{colors.HEADER}CREATING KERNEL JSON...{colors.ENDC}")
+    source = f"""{{
+  "argv": ["{{resource_dir}}/kernel-helper.sh", "python3", "-m", "ipykernel_launcher", "-f", "{{connection_file}}"],
+  "display_name": "{kernel_name}",
+  "language": "python"
+}}"""
+    try:
+        f = open(path + "/kernel.json", "w")
+        f.write(source)
+        f.close()
+    except OSError as e:
+        print(f"{colors.FAIL}failed to write to {path}/kernel.json: {e}{colors.ENDC}", file=sys.stderr)
+        raise JupyterKernelGenException
+    print()
+
+def main():
+    path = None
+    try:
+        # Get the name of the kernel
+        kernel_name = get_kernel_name()
+
+        # Create a directory for the kernel
+        path = create_kernel_dir(kernel_name)
+
+        # Make sure conda or mamba is installed
+        conda_exe = check_for_conda()
+
+        # Figure out which conda environment to use
+        conda_env = get_conda_env()
```

Comment:
I would ask for their conda environment first before asking for their new Kernel name.

- ask for conda env to make into a kernel
- ask for kernel name 
- check to see if ipykernel installed 
- create kernel folder.

Reply:
Yeah I agree. I initially did kernel folder creation earlier because I did my checks for the kernel already existing in that function, but moving that check into kernel name probably makes more sense anyways.